mono/utils/mono-error-internals.h

```diff
@@ -43,10 +43,34 @@ struct _MonoErrorBoxed {
 	MonoImage *image;
 };
 
-#define error_init(error) do {	\
-	((MonoErrorInternal*)(error))->error_code = MONO_ERROR_NONE;	\
-	((MonoErrorInternal*)(error))->flags = 0;	\
-} while (0);
+// FIXME This is duplicated.
+#define ERROR_DECL \
+	MonoError error_value; \
+	MonoError * const error = &error_value; \
+	error_init_internal (error);
```

Comment:
Maybe leave off the trailing semicolon so that callers have to write `ERROR_DECL;` ?

Reply:
Ok either way.